mlrun/model_monitoring/db/tsdb/tdengine/tdengine_connector.py

```diff
@@ -129,6 +138,10 @@ def write_application_event(
         insert_statement.add_batch()
         insert_statement.execute()
 
+    @staticmethod
+    def _convert_to_datetime(key):
+        return datetime.fromisoformat(key) if isinstance(key, str) else key
```

Comment:
Not sure why it would be called `key`.
```suggestion
    def _convert_to_datetime(val):
        return datetime.fromisoformat(val) if isinstance(val, str) else val
```

Reply:
And consider adding type hints :)